src/Catel.MVVM/Catel.MVVM.Shared/Services/UIVisualizerService.wpf.cs

```diff
@@ -170,13 +172,13 @@ protected virtual void HandleCloseSubscription(object window, object data, Event
 
             if (showModal)  // CTL-648 async modal fix
             {
-                var tcs = new TaskCompletionSource<bool?>();
-                HandleCloseSubscription(window, null, (s, e) => tcs.SetResult(e.Result), true);   // complete the task with DialogResult when dialog is closed
-                ShowSystemWindowAsync(window, true);
-                return tcs.Task;
+                // var tcs = new TaskCompletionSource<bool?>();
```

Comment:
And now this code can be simplified a lot by just using this:

return ShowSystemWindowAsync(window, showModal);